### PR TITLE
torch.func backend for implicit FIM

### PR DESCRIPTION
**Pull request by pierrenodet**
supersede #167 (removing dependency of #167 on #164)

### CODE REVIEW

File: nngeometry/backend/torch_func_hessian.py [hunk 1]
```diff
@@ -41,18 +35,20 @@ def get_covariance_matrix(self, examples, layer_collection):
...
 
-        def compute_loss(params, X, y):
-            prediction = torch.func.functional_call(self.model, params, (X,))
-            return self.function(prediction, y)
+        def compute_loss(params, inputs, targets):
+            prediction = torch.func.functional_call(self.model, params, (inputs,))
+            return self.function(prediction, targets)
 
         params_dict = dict(layer_collection.named_parameters(layerid_to_mod))
+        params_dict = {k: v.detach() for k, v in params_dict.items()}
```
Comment (tfjgeorge) [inline nngeometry/backend/torch_func_hessian.py]: why do we need to detach ?

Comment (pierrenodet) [reply nngeometry/backend/torch_func_hessian.py]: otherwise it would consume way more memory, not detaching would allow to differentiate through the hvp/fvp, but I think the use case is niche for nngeometry.
